java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -0,0 +1,94 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import glide.api.commands.SortedSetBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Arguments for {@link SortedSetBaseCommands#zunion}, and {@link
+ * SortedSetBaseCommands#zunionWithScores}
+ *
+ * @see <a href="https://redis.io/commands/zunion/">redis.io</a>
+ */
+public class WeightAggregateOptions {
+    public static final String WEIGHTS_REDIS_API = "WEIGHTS";
+    public static final String AGGREGATE_REDIS_API = "AGGREGATE";
+
+    /**
+     * Option for the method of aggregating scores from multiple sets. This option defaults to SUM if
+     * not specified.
+     */
+    @RequiredArgsConstructor
+    public enum Aggregate {
+        /** Aggregates by summing the scores of each element across sets. */
+        SUM("SUM"),
+        /** Aggregates by selecting the minimum score for each element across sets. */
+        MIN("MIN"),
+        /** Aggregates by selecting the maximum score for each element across sets. */
+        MAX("MAX");
+
+        private final String redisApi;
+
+        public String[] toArgs() {
+            return new String[] {AGGREGATE_REDIS_API, toString()};
+        }
+    }
+
+    /**
+     * Basic interface. Please use one of the following implementations:
+     *
+     * <ul>
+     *   <li>{@link KeyArray}
+     *   <li>{@link WeightedKeys}
+     * </ul>
+     */
+    public interface KeysOrWeightedKeys {
+        String[] toArgs();
+    }
+
+    /** Represents the keys of the sorted sets involved in the aggregation operation. */
+    @RequiredArgsConstructor
+    public static class KeyArray implements KeysOrWeightedKeys {
+        private final String[] keys;
+
+        @Override
+        public String[] toArgs() {
+            return concatenateArrays(new String[] {Integer.toString(keys.length)}, keys);
+        }
+    }
+
+    /**
+     * Represents the mapping of sorted set keys to their multiplication factors. Each factor is used
+     * to adjust the scores of elements in the corresponding sorted set by multiplying them before
+     * their scores are aggregated.
+     */
+    @RequiredArgsConstructor
+    public static class WeightedKeys implements KeysOrWeightedKeys {
+        private final Map<String, Double> keysWeights;
```

Comment:
this is still problematic, we cant pass the same key twice:

```
127.0.0.1:6379> zadd x 1 one 2 two 3 three
(integer) 3
127.0.0.1:6379> zadd y 1 one 2 two 3 three
(integer) 3
127.0.0.1:6379> zunion 2 x y weights 2 3 withscores
1) "one"
2) "5"
3) "two"
4) "10"
5) "three"
6) "15"
```

Reply:
I think we should have something like this:
KeysOrWeightedKeys that holds an array of keys, KeyArray that simply inherits KeysOrWeightedKeys
and WeightedKeys that holds another array but this time of weights

Reply:
we cant accept the same key twice just for one solution

```
127.0.0.1:6379> zunion 2 x y  withscores
1) "one"
2) "2"
3) "two"
4) "4"
5) "three"
6) "6"
```

Reply:
I can have a List[Pair(String, Weights)]

Reply:
yes that would be better
KeyArray remains as it was and KeysOrWeightedKeys will hold a List[Pair(String, Double)]

Reply:
Updated pls take a look.